lib/api/core/models/repositories/repository.js

```diff
@@ -252,6 +313,27 @@ Repository.prototype.persistToDatabase = function (object) {
 };
 
 /**
+ * Delete repository from database according to its id
+ * @param id
+ */
+Repository.prototype.deleteFromDatabase = function (id) {
```

Comment:
need also a deleteFromCache and a delete methods, for data that are in db and in cache.